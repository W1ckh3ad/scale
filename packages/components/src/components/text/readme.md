# scale-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type     | Default     |
| ------------- | -------------- | ------------------------------ | -------- | ----------- |
| `customClass` | `custom-class` | (optional) Text class          | `string` | `''`        |
| `styles`      | `styles`       | (optional) Injected jss styles | `any`    | `undefined` |
| `tag`         | `tag`          | (optional) Text tag            | `string` | `''`        |
| `variant`     | `variant`      | (optional) Text variant        | `string` | `'body'`    |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
