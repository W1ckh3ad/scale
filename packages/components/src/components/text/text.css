.text {
  margin: 0;
  margin-bottom: var(--scl-spacing-2);
}
