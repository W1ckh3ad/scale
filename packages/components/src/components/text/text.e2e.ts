import { newE2EPage } from '@stencil/core/testing';

describe('scale-text', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<scale-text>this is a tag</scale-text>');
    const element = await page.find('scale-text');
    expect(element).toHaveClass('hydrated');
  });
});
