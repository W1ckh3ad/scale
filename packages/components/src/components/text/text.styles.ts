import { JssStyle } from 'jss';

export const styles: JssStyle = {
  text: {
    margin: 0,
    marginBottom: '.5rem', // TODO use token
    // '*::slotted': {
    //   fontSize: 'unset',
    // },
  },
};
