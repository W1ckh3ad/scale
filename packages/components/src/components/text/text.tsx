import { Component, h, Prop, Host } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './text.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';
import { getTheme } from '../../theme/theme';

const variantStyles = () => {
  const result = {};
  const themeVariants = getTheme().type_variants;

  Object.keys(themeVariants).map(key => {
    const name = key.replace('_', '-');
    result[`text--variant-${name}`] = themeVariants[key];
  });
  return result;
};

@Component({
  tag: 'scale-text',
  shadow: true,
})
export class Text implements Base {
  variants: object = variantStyles();

  /** (optional) Text class */
  @Prop() customClass?: string = '';
  /** (optional) Text variant */
  @Prop() variant?: string = 'body';
  /** (optional) Text tag */
  @Prop() tag?: string = '';

  /** (optional) Injected jss styles */
  @Prop() styles?: any = {};
  /** decorator Jss stylesheet */
  @CssInJs('Text', styles) stylesheet: StyleSheet;

  componentWillUpdate() {}
  componentDidUnload() {}

  componentWillRender() {
    // We avoid calling this.stylesheet.addRules(this.variants)
    // on every render for performance reasons
    this.styles = Object.assign(this.styles, this.variants);
  }

  render() {
    const Tag = this.tag || 'p';

    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <Tag class={this.getCssClassMap()}>
          <slot />
        </Tag>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(
      classes.text,
      this.customClass && this.customClass,
      this.variant && classes[`text--variant-${this.variant}`]
    );
  }
}
