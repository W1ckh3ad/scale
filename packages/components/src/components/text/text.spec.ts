import { newSpecPage } from '@stencil/core/testing';
import { Text } from './text';
import { styles } from './text.styles';
import jss from 'jss';

describe('Text', () => {
  let element;
  beforeEach(async () => {
    element = new Text();
    element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Text],
      html: `<scale-text>Label</scale-text>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle css classes', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });
});
