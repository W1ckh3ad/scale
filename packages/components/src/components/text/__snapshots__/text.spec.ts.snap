// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
<scale-text>
  <mock:shadow-root>
    <style>
      .text-0-0-2 {
  margin: 0;
  margin-bottom: .5rem;
}
.text--variant-label-0-0-3 {
  size: 0.75rem;
  weight: 500;
  leading: 1.3333;
  tracking: 0;
}
.text--variant-label-floating-0-0-4 {
  size: 0.625rem;
  weight: 500;
  leading: 1.2;
  tracking: 0;
}
.text--variant-code-0-0-5 {
  size: 0.75rem;
  weight: 400;
  leading: 1.3333;
  tracking: 0;
}
.text--variant-caption-0-0-6 {
  size: 0.75rem;
  weight: 500;
  leading: 1.3333;
  tracking: 0;
}
.text--variant-helper-text-0-0-7 {
  size: 0.75rem;
  weight: 500;
  leading: 1.3333;
  tracking: 0;
}
.text--variant-smaller-0-0-8 {
  size: 0.75rem;
  weight: 500;
  leading: 1.3333;
  tracking: 0;
}
.text--variant-body-0-0-9 {
  size: 1rem;
  weight: 500;
  leading: 1.5;
  tracking: 0;
}
.text--variant-body-long-0-0-10 {
  size: 1rem;
  weight: 500;
  leading: 1.5;
  tracking: 0;
}
.text--variant-body-short-0-0-11 {
  size: 1rem;
  weight: 500;
  leading: 1.25;
  tracking: 0;
}
.text--variant-body-large-0-0-12 {
  size: 1.25rem;
  weight: 500;
  leading: 1.6;
  tracking: 0;
}
.text--variant-h6-0-0-13 {
  size: 1rem;
  weight: 800;
  leading: 1.5;
  tracking: 0;
}
.text--variant-h5-0-0-14 {
  size: 1.25rem;
  weight: 800;
  leading: 1.25;
  tracking: 0;
}
.text--variant-h4-0-0-15 {
  size: 1.5rem;
  weight: 800;
  leading: 1.3333;
  tracking: 0;
}
.text--variant-h3-0-0-16 {
  size: 2rem;
  weight: 800;
  leading: 1.2;
  tracking: 0;
}
.text--variant-h2-0-0-17 {
  size: 2.5rem;
  weight: 800;
  leading: 1.2;
  tracking: 0;
}
.text--variant-h1-0-0-18 {
  size: 3.5rem;
  weight: 800;
  leading: 1.1428571429;
  tracking: 0;
}
.text--variant-h6-mobile-0-0-19 {
  size: 1rem;
  weight: 700;
  leading: 1.5;
  tracking: 0;
}
.text--variant-h5-mobile-0-0-20 {
  size: 1rem;
  weight: 800;
  leading: 1.5;
  tracking: 0;
}
.text--variant-h4-mobile-0-0-21 {
  size: 1.25rem;
  weight: 800;
  leading: 1.25;
  tracking: 0;
}
.text--variant-h3-mobile-0-0-22 {
  size: 1.5rem;
  weight: 800;
  leading: 1.3333;
  tracking: 0;
}
.text--variant-h2-mobile-0-0-23 {
  size: 2rem;
  weight: 800;
  leading: 1.2;
  tracking: 0;
}
.text--variant-h1-mobile-0-0-24 {
  size: 2.5rem;
  weight: 800;
  leading: 1.2;
  tracking: 0;
}
.text--variant-collapsible-0-0-25 {
  size: 1.25rem;
  weight: 800;
  leading: 1.25;
  tracking: 0;
  margin-bottom: 0;
}
    </style>
    <p class="text--variant-body-0-0-9 text-0-0-2">
      <slot></slot>
    </p>
  </mock:shadow-root>
  Label
</scale-text>
`;
